src/bootstrap/src/core/build_steps/gcc.rs

```diff
@@ -0,0 +1,36 @@
+//! Retrieves information related to GCC backend.
+
+use crate::builder::Cargo;
+
+/// This retrieves the gccjit sha we use.
+pub(crate) fn detect_gccjit_sha() -> String {
+    // FIXME: This is absolutely ugly. Please find another way to do that.
+    let content =
+        include_str!("../../../../ci/docker/host-x86_64/dist-x86_64-linux/build-gccjit.sh");
+    if let Some(commit) = content
+        .lines()
+        .filter_map(|line| line.strip_prefix("GIT_COMMIT=\""))
+        .filter_map(|commit| commit.strip_suffix("\""))
+        .next()
+    {
+        if !commit.is_empty() {
+            return commit.to_string();
+        }
+    }
+
+    eprintln!("error: could not find commit hash for downloading libgccjit");
+    panic!();
+}
+
+pub fn set_gccjit_library_path(config: &crate::Config, cargo: &mut Cargo) {
+    if config.gcc_download_gccjit {
+        let libgccjit_path = config.libgccjit_folder(&detect_gccjit_sha());
+        let mut library_path = std::env::var("LD_LIBRARY_PATH").unwrap_or_default();
+        if !library_path.is_empty() {
+            library_path.push(':');
+        }
+        library_path.push_str(&libgccjit_path.display().to_string());
+        cargo.env("LD_LIBRARY_PATH", &library_path);
+        cargo.env("LIBRARY_PATH", library_path);
```

Comment:
I think this approach is wrong.

1) Path name can be different depending on the platform. https://github.com/rust-lang/rust/blob/cb3752d20e0f5d24348062211102a08d46fbecff/src/bootstrap/src/utils/dylib.rs#L3-L17

2) This will either override other values or be overridden by other values from https://github.com/rust-lang/rust/blob/cb3752d20e0f5d24348062211102a08d46fbecff/src/bootstrap/src/utils/helpers.rs#L73-L81

3) At some point, we need these values to be returned from `Builder::rustc_lib_paths`. If we handle the logic of finding the library (with `if config.gcc_download_gccjit` obviously) there, it will automatically be set with the appropriate path name for the commands which needs this path.

Reply:
Sounds good to me!